<launch>

    <!-- Configurable arguments -->
    <arg name="robot_name" default="sam"/>
	<arg name="base_frame" default="/$(arg robot_name)/base_link" />
	<arg name="utm_frame" default="utm" />
	<arg name="max_depth" default="20" />
	<arg name="min_altitude" default="5" />
	<arg name="emergency_trials_before_giving_up" default=10 />

	<!-- leave empty string to disable path planner -->
	<arg name="path_planner_name" default="/interp1d" />

	<!-- Croatia, biograd coast, lat lon zone band, all four from there -->
    <arg name="utm_zone" default="33"/>
    <arg name="utm_band" default="T"/>

	<!-- topics -->
    <arg name="plandb_topic" default="imc/plan_db"/>
    <arg name="plan_control_topic" default="imc/plan_control"/>
    <arg name="estimated_state_topic" default="imc/estimated_state"/>
    <arg name="plan_control_state_topic" default="imc/plan_control_state"/>
    <arg name="vehicle_state_topic" default="imc/vehicle_state"/>
    <arg name="abort_topic" default="imc/abort"/>
    <arg name="gps_fix_topic" default="core/gps"/>
    <arg name="altitude_topic" default="core/dvl"/>


	<!-- Action Servers -->
    <arg name="action_namespace" default="ctrl/wp_depth_action_planner"/>
    <arg name="emergency_action_namespace" default="ctrl/emergency_surface_action"/>


	<node name="sam_bt" pkg="smarc_bt" type="sam_bt.py" output="screen" ns="$(arg robot_name)">
		<param name="base_frame" value="$(arg base_frame)" />
		<param name="utm_frame" value="$(arg utm_frame)" />
		<param name="path_planner_name" value="$(arg path_planner_name)" />

		<param name="max_depth" value="$(arg max_depth)" />
		<param name="min_altitude" value="$(arg min_altitude)" />
		<param name="emergency_trials_before_giving_up" value="$(arg emergency_trials_before_giving_up)" />

		<param name="utm_zone" value="$(arg utm_zone)" />
		<param name="utm_band" value="$(arg utm_band)" />

		<param name="plandb_topic" value="$(arg plandb_topic)" />
		<param name="plan_control_topic" value="$(arg plan_control_topic)" />
		<param name="estimated_state_topic" value="$(arg estimated_state_topic)" />
		<param name="plan_control_state_topic" value="$(arg plan_control_state_topic)" />
		<param name="vehicle_state_topic" value="$(arg vehicle_state_topic)" />
		<param name="abort_topic" value="$(arg abort_topic)" />

		<param name="action_namespace" value="$(arg action_namespace)" />
		<param name="emergency_action_namespace" value="$(arg emergency_action_namespace)" />
	</node>

    
</launch>
